### PR TITLE
Domaincheck analyzer, closes #1682

### LINKED ISSUE #1682: [Analyzer] Domaincheck
## Name
Domaincheck

## Link

https://github.com/zuzazuza/domaincheck

## Type of analyzer
domain only

## Why should we use it
cool tool to check DNS related problems

## Possible implementation
integrate the open source tool: https://github.com/zuzazuza/domaincheck

### PR DESCRIPTION
**Pull request by g4ze**
closes #1682 

# Description

Please include a summary of the change and link to the related issue.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ ] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [ ] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ ] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require any API key), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
    - [ ] If the plugin interacts with an external service, I have created an attribute called precisely `url` that contains this information. This is required for Health Checks. 
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have added that raw JSON sample to the `MockUpResponse` of the `_monkeypatch()` method. This serves us to provide a valid sample for testing.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [ ] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

**Comment by g4ze** [comment]
```
>>> from domaincheck import domaincheck
>>> result = domaincheck.main(["google.com"],resolver="8.8.8.8")
performAllChecks2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
lookup2: resolver= 8.8.8.8
>>> print(result)
```

@mlodic this script works perfectly fine inside the docker container, but for some reason, when using the same logic inside the analyzer this is shown:
image

logs:

```
2024-07-23 18:42:03,722 - api_app.views - analyze_multiple_observables - INFO - received analyze_multiple_observables from user admin
2024-07-23 18:42:03,730 - api_app.serializers.job - validate - DEBUG - before attrs: {'user': <SimpleLazyObject: <User: admin>>, 'delay': 0, 'is_sample': False, 'tlp': 'AMBER', 'analyzers_requested': [<AnalyzerConfig: Domaincheck>], 'connectors_requested': <PythonConfigQuerySet []>, 'tags_labels': <generator object _AbstractJobCreateSerializer.validate_tags_labels at 0x7f72d6fd3040>, 'scan_mode': 1, 'investigation': None, 'observable_name': 'google.com'}
2024-07-23 18:42:03,764 - api_app.serializers.job - validate - DEBUG - after attrs: {'user': <SimpleLazyObject: <User: admin>>, 'delay': 0, 'is_sample': False, 'tlp': 'AMBER', 'analyzers_requested': [<AnalyzerConfig: Domaincheck>], 'connectors_requested': <PythonConfigQuerySet []>, 'scan_mode': 1, 'investigation': None, 'observable_name': 'google.com', 'observable_classification': ObservableTypes.DOMAIN, 'md5': '1d5920f4b44b27a802bd77c4f0536f5a', 'scan_check_time': None, 'analyzers_to_execute': [<AnalyzerConfig: Domaincheck>], 'connectors_to_execute': [], 'visualizers_to_execute': [], 'warnings': [], 'tags': <generator object _AbstractJobCreateSerializer.validate_tags_labels at 0x7f72d6fd3040>}
2024-07-23 18:42:03,807 - api_app.serializers.job - create - INFO - Job 107 created
2024-07-23 18:42:03,807 - api_app.serializers.job - create - INFO - Sending task for job 107
2024-07-23 18:42:03,830 - api_app.views - analyze_multiple_observables - INFO - finished analyze_multiple_observables from user admin
```
Am I missing something obvious here?

**Comment by mlodic** [comment]
from your log it seems that your main function is delivering results to the standard output instead of as a return to the function itself so "results" is never populated. Otherwise you should not see anything in the shell logs

**Comment by mlodic** [comment]
maybe enable json result? https://github.com/intelowlproject/domaincheck/blob/main/domaincheck/domaincheck.py#L388

**Comment by g4ze** [comment]
I'm not sure I understood what you meant. In the python shell, the result `variable` does contain a proper json result. when I do` print(result)` in the shell, it does print the complete result. 
secondly, the only output supported(and by default) in the new package is json. so it is inevitable for the funnction to return anything else.
The result in the python shell is always populated and the main function does return the result in a json format.
Moreover, I have put some start and end logs in the analyzer class, which dont seem to be working, as seen in the api logs. So my guess is that the control flow isint even reaching the analyzer class in this case. Please let me know if it makes sense.

File: api_app/analyzers_manager/observable_analyzers/domaincheckanalyzer.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from api_app.analyzers_manager import classes
+
+logger = logging.getLogger(__name__)
+
+
+class DomainCheck(classes.ObservableAnalyzer):
+    resolver: str = "8.8.8.8"
+
+    def run(self):
+        logger.info(f"Running DomainCheck for {self.observable_name}")
```
Comment (mlodic) [inline api_app/analyzers_manager/observable_analyzers/domaincheckanalyzer.py:14]: you don't see this log right?

Comment (g4ze) [reply api_app/analyzers_manager/observable_analyzers/domaincheckanalyzer.py:14]: Yes

**Comment by mlodic** [comment]
abandoned cause main code is too raw

**Pull request closed without merging**